zipkin-server/src/test/java/zipkin2/server/internal/ITZipkinGrpcCollector.java

```diff
@@ -80,30 +79,33 @@ class ITZipkinGrpcCollector {
   }
 
   @Test void report_emptyIsOk() throws IOException {
-
     callReport(new ListOfSpans.Builder().build());
   }
 
-  ReportResponse callReport(ListOfSpans spans) throws IOException {
-    Buffer requestBody = new Buffer();
-    requestBody.writeByte(0 /* compressedFlag */);
-    Buffer encodedMessage = new Buffer();
-    ListOfSpans.ADAPTER.encode(encodedMessage, spans);
-    requestBody.writeInt((int) encodedMessage.size());
-    requestBody.writeAll(encodedMessage);
-
-    Response response = client.newCall(new Request.Builder()
-      .url(url(server, "/zipkin.proto3.SpanService/Report"))
-      .addHeader("te", "trailers")
-      .post(RequestBody.create(requestBody.snapshot(), MediaType.get("application/grpc")))
-      .build())
-      .execute();
-
-    BufferedSource responseBody = response.body().source();
-    assertThat((int) responseBody.readByte()).isEqualTo(0); // uncompressed
-    long encodedLength = responseBody.readInt() & 0xffffffffL;
-
-    return ReportResponse.ADAPTER.decode(responseBody);
+  void callReport(ListOfSpans spans) throws IOException {
+    try (Buffer requestBody = new Buffer(); Buffer encodedMessage = new Buffer()) {
+      requestBody.writeByte(0 /* compressedFlag */);
+
+      ListOfSpans.ADAPTER.encode(encodedMessage, spans);
+      requestBody.writeInt((int) encodedMessage.size());
+      requestBody.writeAll(encodedMessage);
+
+      Request request = new Request.Builder()
+        .url(url(server, "/zipkin.proto3.SpanService/Report"))
+        .addHeader("te", "trailers")
+        .post(RequestBody.create(requestBody.snapshot(), MediaType.get("application/grpc")))
+        .build();
+      try (Response response = client.newCall(request).execute();
+           BufferedSource responseBody = response.body().source()) {
+
+        assertThat((int) responseBody.readByte()).isEqualTo(0); // uncompressed
+        long encodedLength = responseBody.readInt() & 0xffffffffL;
+        assertThat(encodedLength).isEqualTo(encodedMessage.size());
```

Comment:
I must be missing something but I don't quite understand why the response length is equal to the request message length. Shouldn't it be 0 since the response proto is empty?

Reply:
yeah really bad of me to match the two words without thinking. we exhausted the encoded message so it is zero.

Reply:
thanks, PTAL!